uwblueprint/community-fridge-kw
# feat: Update Volunteer Model with Pending/Approved ENUM

## Brief description. What is this change? 

### [Update Volunteer Model with Pending/Approved ENUM + update BE types](https://www.notion.so/uwblueprintexecs/Update-Volunteer-Model-with-Pending-Approved-ENUM-update-BE-types-12298bcb9eaf47a4a11851b31471777e)
### [Update Volunteer Service, Route, and Middleware](https://www.notion.so/uwblueprintexecs/39bf3fa8ccce4beb93205438181ec01a?v=455e82ee04d9425d99a8322a3395d1a7&p=10b7eac7b1e74a0e8e8f21436de30639)



## Implementation description. How did you make this change?
* Updated `volunteer.model.ts` to include a `status` enum
* Updated `VolunteerDTO` type
* Updated services in `volunteerService.ts` to return status
* Updated `createVolunteer` in `authRoutes.ts` to include status
* Created a validator for ```volunteerService```
* Cleaned up the ```get``` volunteer endpoint because it was breaking for us



## Steps to test
(Sorry it's so long :))
1. Drop your tables to rebuild with the new status column on volunteers and ensure it is there.

pgAdmin: ![image](https://user-images.githubusercontent.com/69697180/152657928-0977823c-983d-4b5f-91cb-73f7fc6faa01.png)

2. Create users on ```POST localhost:5000/auth/register``` with an example payload:
```
{
    "firstName": "Harry",
    "lastName": "Potter",
    "email": "tedd2ybears1233@gmail.ca",
    "role": "Volunteer",
    "phoneNumber": "5555555555",
    "password": "Testing1234!",
    "status": "Pending"
}
```
Play around with creating both volunteers and donors so ```userId``` and ```id``` is different to validate testing. Make sure you see the new users in pgadmin.
3. Test validators
```
{
    "firstName": "Harry",
    "lastName": "Potter",
    "email": "tedd2ybears1233@gmail.ca",
    "role": "Volunteer",
    "phoneNumber": "5555555555",
    "password": "Testing1234!",
    "status": "Jigglypuff"
}
```
A payload like this should return an error message: ```The status is not a Status```. 
4. Test update by id at ```PUT localhost:5000/volunteers/<id>``` with a payload like
```
{
    "status": "Approved"
}
```
Ensure that the volunteer is updated in pgadmin.
5. Test update by userId at ```PUT localhost:5000/volunteers/?userId=<userId>``` with a payload like
```
{
    "status": "Rejected"
}
```
Ensure that the volunteer is updated in pgadmin.
6. Try both id and userId ```PUT localhost:5000/volunteers/<id>?userId=<userId>``` with any payload and an error should appear: ```Cannot update by both volunteerId and userId```
7. If neither id or userId is in your PUT request, the error: ```Must supply id as request parameter.``` should appear.

## Checklist
